src/main/java/messaging/MessageInteractor.java

```diff
@@ -0,0 +1,54 @@
+package messaging;
+
+import database.RoomDBGateway;
+import models.MessageModel;
+import models.RoomDBModel;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class MessageInteractor implements MessageInputBoundary {
+    private final MessageOutputBoundary presenter;
+    private final RoomDBGateway database;
+
+
+    public MessageInteractor(RoomDBGateway database, MessageOutputBoundary presenter) {
+        this.presenter = presenter;
+        this.database = database;
+    }
+
+    /**
+     * Handles the user request for sending a message.
+     */
+    @Override
+    public void handleSendMessage(MessageModel request) {
+        RoomDBModel room = database.getARoom(request.getRoomID());
+
```

Comment:
database.getARoom(request.getRoomID()); this tries to get room id from the database, what if the room id does not exist in the database due to some error after they joining? Maybe you can include this if (room == null)  or you can use try(success) catch to handle exceptions

Reply:
ok, sounds good